lib/Parser/parse.cpp

```diff
@@ -2076,6 +2117,556 @@ ParseNodePtr Parser::ParseMetaProperty(tokens metaParentKeyword, charcount_t ich
+
+    if (importEntry == nullptr)
+    {
+        OutOfMemory();
+    }
+
+    importEntry->importName = importName;
+    importEntry->localName = localName;
+    importEntry->moduleRequest = moduleRequest;
+    importEntry->varDecl = declNode;
+
+    importEntryList->Prepend(*importEntry);
+}
+
+void Parser::AddModuleExportEntry(ModuleExportEntryList* exportEntryList, IdentPtr importName, IdentPtr localName, IdentPtr exportName, IdentPtr moduleRequest)
+{
+    ModuleExportEntry* exportEntry = Anew(&m_nodeAllocator, ModuleExportEntry);
+
+    if (exportEntry == nullptr)
+    {
+        OutOfMemory();
+    }
+
+    exportEntry->importName = importName;
+    exportEntry->localName = localName;
+    exportEntry->exportName = exportName;
+    exportEntry->moduleRequest = moduleRequest;
+
+    exportEntryList->Prepend(*exportEntry);
+}
+
+void Parser::AddModuleLocalExportEntry(ParseNodePtr varDeclNode)
+{
+    Assert(varDeclNode->nop == knopVarDecl || varDeclNode->nop == knopLetDecl || varDeclNode->nop == knopConstDecl);
+
+    IdentPtr localName = varDeclNode->sxVar.pid;
+    varDeclNode->sxVar.sym->SetIsModuleExportStorage(true);
+
+    AddModuleExportEntry(EnsureModuleLocalExportEntryList(), nullptr, localName, localName, nullptr);
+}
+
+template<bool buildAST>
+void Parser::ParseImportClause(ModuleImportEntryList* importEntryList, bool parsingAfterComma)
+{
+    bool parsedNamespaceOrNamedImport = false;
+
+    switch (m_token.tk)
+    {
+    case tkID:
+        // This is the default binding identifier.
+
+        // If we already saw a comma in the import clause, this is a syntax error.
+        if (parsingAfterComma)
+        {
+            Error(ERRsyntax);
+        }
+
+        if (buildAST)
+        {
+            IdentPtr localName = m_token.GetIdentifier(m_phtbl);
+            IdentPtr importName = wellKnownPropertyPids.default;
+            ParseNodePtr declNode = CreateModuleImportDeclNode(localName);
+
+            AddModuleImportEntry(importEntryList, importName, localName, nullptr, declNode);
+        }
+        
+        break;
+    
+    case tkLCurly:
+        // This begins a list of named imports.
+        ParseNamedImportOrExportClause<buildAST>(importEntryList, nullptr, false);
+
+        parsedNamespaceOrNamedImport = true;
+        break;
+
+    case tkStar:
+        // This begins a namespace import clause.
+        // "* as ImportedBinding"
+
+        // Token following * must be the identifier 'as'
+        m_pscan->Scan();
+        if (m_token.tk != tkID || wellKnownPropertyPids.as != m_token.GetIdentifier(m_phtbl))
+        {
+            Error(ERRsyntax);
+        }
+
+        // Token following 'as' must be a binding identifier.
+        m_pscan->Scan();
+        ChkCurTokNoScan(tkID, ERRsyntax);
+
+        if (buildAST)
+        {
+            IdentPtr localName = m_token.GetIdentifier(m_phtbl);
+            IdentPtr importName = CreatePid(L"*", 1);
```

Comment:
Added wellKnownPropertyPids._star for this (and wellKnownPropertyPids._starDefaultStar for the other one.